src/librustdoc/clean/mod.rs

```diff
@@ -388,14 +388,39 @@ impl<'tcx> Clean<Type> for ty::ProjectionTy<'tcx> {
         let trait_ = lifted.trait_ref(cx.tcx).clean(cx);
         let self_type = self.self_ty().clean(cx);
         Type::QPath {
-            name: cx.tcx.associated_item(self.item_def_id).name,
+            assoc: Box::new(self.clean(cx)),
             self_def_id: self_type.def_id(&cx.cache),
             self_type: box self_type,
             trait_,
         }
     }
 }
 
+impl<'tcx> Clean<PathSegment> for ty::ProjectionTy<'tcx> {
```

Comment:
Can you please make this a regular function? I'm trying to reduce usage of `Clean` since it's really confusing when there are multiple impls of it for the same type (and there are a lot of other downsides).